### DISCUSSION TITLE
Can we use matrices for uncertain parameters? Not mentioned in the documentation!

### DISCUSSION THREAD

**Discussion by TourajFarahmand-AI** [do-mpc/do-mpc#248] [Q&A]
Hello do-mpc team,
I tried to use matrices for parameters but it keeps giving me errors when I call set_uncertainty_values. I could not find any clarification in the documentation.

Thanks,
Touraj.

**Comment by 4flixt** [comment]
Hi Touraj,

the values for the uncertain parameter must have the same shape as for the [introduced variables in the model](https://www.do-mpc.com/en/latest/api/do_mpc.model.Model.p.html).

Best
4flixt

**Reply by 4flixt** [reply]
Actually on second thought: Only scalars are possible. It doesn't really make sense to pass vectors or matrices to [mpc.set_unsertainty_values](https://www.do-mpc.com/en/latest/api/do_mpc.controller.MPC.set_uncertainty_values.html) because the method wouldn't be able to create permutations of these.

**Reply by TourajFarahmand-AI** [reply]
Thanks @4flixt since dimension is accepted in the definition of parameters I thought it is somehow supported and I was trying to test the variation of state-space matrices A, B and C. I guess I have to write state space equations as  separate equations which is fine.
